### ISSUE TITLE
Data Source Identifier app

### ISSUE THREAD

**Issue by josh-chamberlain** [Police-Data-Accessibility-Project/data-source-identification#109]
## Context
- Part of https://github.com/Police-Data-Accessibility-Project/data-source-identification/issues/108

## Requirements

Source collectors hunt for data sources. The Data Source Identifier handles them in batches, sorting through them and submitting them to the Data Sources API or setting them aside for human labeling.

- [ ] An endpoint which accepts a batch of URLs, each with the metadata: `agency` `record_type` `relevance`
  - other metadata from the data sources schema can be accepted
- [ ] The app should create a log of the batch, with the strategy used to generate it and basic facts about it
  - builds on the concept of batches: https://github.com/Police-Data-Accessibility-Project/data-source-identification/issues/96
- [ ] The app should filter out sources which are not relevant using our ML models
  - if `relevance` or any other property is submitted for a URL, it means the process upstream is certain and we should bypass that step of identification
- [ ] The app should attempt to find more metadata about each URL in the batch
  - https://github.com/Police-Data-Accessibility-Project/data-sources-app/issues/550
  - minimum submission requirements are `agency` and `record_type`
- [ ] If the app can't identify the source sufficiently, it should be sent to LabelStudio for human identification
- [ ] The app should add to the table of "missingness": sources we have looked for and could not find
  - basically a combination of `location`/`agency` and `record type`, with the last date we checked and maybe a batch ID or the method used 
- [ ] for sources which have the minimum required metadata, they should be submitted to the data sources API for approval. 
  - We may want a separate approval queue because of the scale; maybe batches are approved one at a time.

## Testing
- We should have some.

## Open questions
- [ ] should we set up a new, simple API and database? If so, what do we want to keep from the data sources API/db?
- [ ] how does this API relate to the data sources API?
- [ ] how are all the relevant schemae kept in sync?

**Comment by josh-chamberlain** [comment]
To proceed on this issue, in rough order:

1. decide where the `batch` and `url` tables should be
  - currently `batch` = our own db and `url` = huggingface datasets; do we need to adjust this?
2. make a plan for cross-referencing logic (submitting to data sources db or labelstudio, interacting with huggingface)
3. make the database / tables
4. make a namespace for data source ID API endpoints (and the endpoints)
5. make sure we are labeling batches and urls with success rate/outcome

**Comment by maxachis** [comment]
@josh-chamberlain I'll keep my questions in a fun little checkbox format so I can mark off when we have them

- [ ] To confirm, we did indeed decide on having a separate API and Database on 12/17, correct? So the main description is a bit out of date?

**Comment by josh-chamberlain** [comment]
@maxachis correct! I adjusted the answers to all 3 open questions
